ui/src/components/WorkerDetailsCard/index.jsx

```diff
@@ -1,52 +1,189 @@
+        throw new Error('User credentials not found. Please log in.');
+      }
+
+      const wm = new WorkerManager({
+        rootUrl: window.env.TASKCLUSTER_ROOT_URL,
+        credentials: user.credentials,
+        authorizedScopes: [
+          `worker-manager:remove-worker:${taskQueueId}/${workerGroup}/${workerId}`,
+        ],
+      });
+
+      await wm.removeWorker(taskQueueId, workerGroup, workerId);
+      this.setState({
+        dialogState: {
+          ...this.state.dialogState,
+          open: false,
+        },
+      });
+    } catch (error) {
+      this.handleDialogActionError(error);
+    }
+  };
+
+  handleDialogActionError = error => {
+    this.setState({
+      dialogState: {
+        ...this.state.dialogState,
+        error,
+      },
+    });
+  };
+
+  handleDialogActionClose = () => {
+    this.setState({
+      dialogState: {
+        ...this.state.dialogState,
+        error: null,
+        open: false,
+      },
+    });
+  };
+
   render() {
     const {
-      worker: { quarantineUntil, firstClaim, lastDateActive },
+      worker: {
+        quarantineUntil,
+        firstClaim,
+        lastDateActive,
+        providerId,
+        taskQueueId,
+        workerGroup,
+        workerId,
+      },
     } = this.props;
+    const {
+      dialogState: { open, error, title, confirmText, body },
+    } = this.state;
+    const iconSize = 16;
 
     return (
-      <List>
-        <ListItem>
-          <ListItemText
-            primary="Last Active"
-            secondary={
-              lastDateActive ? <DateDistance from={lastDateActive} /> : 'n/a'
-            }
-          />
-        </ListItem>
-        <ListItem>
-          <ListItemText
-            primary="First Claim"
-            secondary={<DateDistance from={firstClaim} />}
-          />
-        </ListItem>
-        <ListItem>
-          <ListItemText
-            primary="Quarantine Until"
-            secondary={
-              quarantineUntil
-                ? format(parseISO(quarantineUntil), 'yyyy/MM/dd')
-                : 'n/a'
-            }
-          />
-        </ListItem>
-      </List>
+      <Fragment>
+        <List>
+          <ListItem>
+            <ListItemText
+              primary="Last Active"
+              secondary={
+                lastDateActive ? <DateDistance from={lastDateActive} /> : 'n/a'
+              }
+            />
+          </ListItem>
+          <ListItem>
+            <ListItemText
+              primary="First Claim"
+              secondary={<DateDistance from={firstClaim} />}
+            />
+          </ListItem>
+          <ListItem>
+            <ListItemText
+              primary="Quarantine Until"
+              secondary={
+                quarantineUntil
+                  ? format(parseISO(quarantineUntil), 'yyyy/MM/dd')
+                  : 'n/a'
+              }
+            />
+          </ListItem>
+          {providerId !== NULL_PROVIDER ? (
+            <ListItem>
+              <Button
```

Comment:
one suggestion here: if you switch to `ui/src/components/Button`  you will be able to use `requiresAuth` prop, which will disable button if user is not authenticated, and you will not have to do the check on line 68-70